packagedb/to_purl.py

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# purldb is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/purldb for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+from django.db import router
+from drf_spectacular.utils import OpenApiParameter, extend_schema
+from packageurl.utils import get_golang_purl
+from rest_framework import status
+from rest_framework import viewsets
+from rest_framework.response import Response
+
+from packagedb.serializers import GoLangPurlResponseSerializer
+from packagedb.serializers import GoLangPurlSerializer
+from rest_framework import routers
+
+
+@extend_schema(
+    parameters=[
+        OpenApiParameter("go_package", str, "query", description="go import package"),
+    ],
+    responses={200: GoLangPurlResponseSerializer()},
+)
+class GolangPurlViewSet(viewsets.ViewSet):
+    """
+    Return a ``golang_purl`` from a standard go import string 
```

Comment:
This is either an import or a go.mod string. You should give examples as the API user needs help.